modules/product/src/VariationFieldInjection.php

```diff
@@ -0,0 +1,171 @@
+    return $this->fieldDefinitions[$variation_type_id];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRenderableFields(ProductVariationInterface $variation, $view_mode = 'default') {
+    $injectable_fields = [];
+
+    $field_definitions = $this->getFieldDefinitions($variation->bundle());
+
+    $view_display = $this->entityTypeManager
+      ->getStorage('entity_view_display')
+      ->load("commerce_product_variation.{$variation->bundle()}.$view_mode");
+
+    if (!$view_display) {
+      // Use default if one was not available.
+      /** @var \Drupal\Core\Entity\Entity\EntityViewDisplay $view_display */
+      $view_display = commerce_get_entity_display('commerce_product_variation', $variation->bundle(), 'view');
+    }
+
+    $view_builder = $this->entityTypeManager->getViewBuilder('commerce_product_variation');
+
+    foreach ($field_definitions as $field_name => $field_definition) {
+      if ($view_display->getComponent($field_name)) {
+        $content = $view_builder->viewField($variation->$field_name, $view_mode);
+        $content['#attributes']['class'][] = $this->baseCssClass();
+
+        $content['#attributes']['class'][] = $this->variationFieldCssClass('variation_' . $field_name, $variation);
+        $content['#ajax_injection_class'] = $this->variationFieldCssClass('variation_' . $field_name, $variation);
+        $injectable_fields['variation_' . $field_name] = $content;
+      }
+    }
+
+    return $injectable_fields;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addAjaxCommands(AjaxResponse $response, ProductVariationInterface $variation, $view_mode = 'default') {
+    foreach ($this->getRenderableFields($variation, $view_mode) as $field_name => $injection_field) {
+      $response->addCommand(new ReplaceCommand('.' . $injection_field['#ajax_injection_class'], $injection_field));
+    }
+
+    $this->moduleHandler->invokeAll('commerce_product_variation_field_injection', [
+      $response,
+      $variation,
+      $view_mode,
+    ]);
+  }
+
+  /**
+   * The allowed base field definitions for injection.
+   *
+   * @return array
+   *   An array of base field names.
+   */
+  protected function allowedBaseFields() {
+    return ['title', 'sku', 'price'];
+  }
+
+  /**
+   * The base CSS class all injected fields will have.
+   *
+   * This is also the base block class for injectable field classes that can be
+   * accessed via AJAX, follow BEM concepts.
+   *
+   * @return string
+   *   The CSS class.
+   */
+  protected function baseCssClass() {
+    return 'product--variation-field';
+  }
+
+  /**
+   * The CSS class used to identify the field with AJAX callbacks.
+   *
+   * @param string $field_name
+   *   The field name.
+   * @param \Drupal\commerce_product\Entity\ProductVariationInterface $variation
+   *   The product variation.
+   *
+   * @return string
+   *   The CSS class.
+   */
+  protected function variationFieldCssClass($field_name, ProductVariationInterface $variation) {
```

Comment:
Like I've already commented it at baseCssClass(): It's really an edge case. But why not keep and only rename it?